### ISSUE TITLE
Write test and illustrate use of autodiff gradient

### ISSUE THREAD

**Issue by redeboer** [ComPWA/tensorwaves#442]
There are currently no unit tests for [`gradient_creator()`](https://tensorwaves.readthedocs.io/en/0.4.5/api/tensorwaves.estimator.html#tensorwaves.estimator.gradient_creator) and I don't trust the implementation for JAX -- it calls `jax.grad()` on a `ParametrizedFunction`, not the internal JAX function.

It would also be nice to illustrate the creation of a gradient for a function in a notebook. (E.g., a simple demo of [error propagation with autodiff](https://colab.research.google.com/github/cranmer/stats-ds-book/blob/master/book/error-propagation/error_propagation_with_jax.ipynb).)
